lib/src/sai_redis_record.cpp

```diff
@@ -0,0 +1,105 @@
+#include "sai_redis.h"
+
+std::string getTimestamp()
+{
+    SWSS_LOG_ENTER();
+
+    char buffer[64];
+    struct timeval tv;
+
+    gettimeofday(&tv, NULL);
+
+    size_t size = strftime(buffer, 64 ,"%Y-%m-%d.%T.", localtime(&tv.tv_sec));
+
+    sprintf(&buffer[size],"%06ld", tv.tv_usec);
+
+    return std::string(buffer);
+}
+
+// currenly true by default for debugging
+volatile bool g_record = true;
+
+std::ofstream recording;
+
+void recordLine(std::string s)
+{
+    SWSS_LOG_ENTER();
+
+    if (recording.is_open())
+    {
+        recording << getTimestamp() << "," << s << std::endl;
+
+        // performance hit
+        recording.flush();
+    }
+}
+
+std::string recfile = "dummy.rec";
```

Comment:
currently every start recording user gets new file name based on timestamp, later we could provide some interface to add abbility for user to specify desired filename, this is not priority right now